### PR TITLE
feat: add regex entity extractor

### PR DESCRIPTION
**Pull request by lxobr**
## Description

- Created a new RegexEntityExtractor that uses regex patterns to identify entities like emails, URLs, and dates in text
-  Implemented a JSON-based configuration system to add or modify entity types without changing code
- Built a  separate RegexEntityConfig class to handle loading and processing of entity configurations
- Added test suite covering all entity types and edge cases
## DCO Affirmation
I affirm that all code in every commit of this pull request conforms to the terms of the Topoteretes Developer Certificate of Origin

## Summary by CodeRabbit

- **New Features**
	- Introduced a new regex-based extraction capability that uses configurable patterns and description templates to identify common entities such as emails, phone numbers, URLs, dates, and more.
- **Tests**
	- Added comprehensive tests to validate the extraction functionality across standard scenarios and edge cases for reliable text analysis.

### CODE REVIEW

**Comment by soobrosa** [comment]
Adding a reasonable multilingual company name pattern would be outstanding but I think this is good to go!

**Review by soobrosa — APPROVED** [review]
LGTM

File: cognee/tasks/entity_completion/entity_extractors/regex_entity_extractor.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+from cognee.infrastructure.entities.BaseEntityExtractor import BaseEntityExtractor
+from cognee.modules.engine.models import Entity
+from cognee.root_dir import get_absolute_path
+from cognee.tasks.entity_completion.entity_extractors.regex_entity_config import RegexEntityConfig
+
+logger = logging.getLogger("regex_entity_extractor")
+
+
+class RegexEntityExtractor(BaseEntityExtractor):
```
Comment (borisarzentar) [inline cognee/tasks/entity_completion/entity_extractors/regex_entity_extractor.py:12]: File names don't match the class name.
